### ISSUE TITLE
fp8 native artifact compilation takes 100+ s for a two-estimator model on nv0

### ISSUE THREAD

**Issue by nm-z** [nm-z/recipe-dev#656]
Public `recipe.train().optimizer(adamw).lr(0.001).seed(71298284147480).epochs(1).log(all).stop(0.0).fp(8).save(&bundle).run(&model, &data)` on backend `nv0` spends over 100 seconds compiling native artifacts before the first epoch runs. The epoch itself takes 43 ms and the warm run of the same program takes 3.4 s, so the entire cost is the one-time native compilation of `fp8` kernels.

This was found while verifying the packets on #497. It is not the XGBoost preparation path that #497 names, so it needs its own issue.

## Program

```rust
use recipe::*;

fn main() {
	let bundle = "/home/nate/.cache/recipe/packets/issue-497b.ogdl";
	let data = recipe.data("data/temporal/chronological_splits")
		.target("target")
		.split(0.8);
	let model = recipe.model()
		.xgbst().log().qi(4).0
		.cbst().tan().norm(batch).qi(5).0
		.loss(mae);
	let report = recipe.train()
		.optimizer(adamw)
		.lr(0.001)
		.seed(71298284147480)
		.epochs(1)
		.log(all)
		.stop(0.0)
		.fp(8)
		.save(&bundle)
		.run(&model, &data);
	assert!(report.final_loss().is_finite());
}
```

## Measurement

`recipe --device nv0 issue-497b.rs` on archy nv0 (Tesla M60, die 3), fresh `HOME` so the native cache is empty, with other agents active on the host:

| run | cold wall | warm wall |
|---|---|---|
| 1 | 96.4 s | 4.0 s |
| 2 | 103.1 s | 7.1 s |
| 3 | 108.8 s | 3.4 s |
| 4 | 127.4 s | 5.6 s |

Epoch time is 0.59 s predicted and 43 ms measured in every run, and the bundle is 307,790 bytes with an unchanged SHA-256 across runs.

## Where the time goes

A cold run compiles six native artifacts, one after another, and every one is a cache miss at `arithmetic=fp8`. From `recipe.log` and the artifact directory timestamps of a single instrumented cold run:

| artifact | rows | loss | PTX size | compile |
|---|---|---|---|---|
| 7d77ed57147ebd44 | 14400 | mse | 760 KB | 5 s |
| 71be3413348a0d38 | 14400 | none | 1460 KB | 7 s |
| 1b628e6be373ee2b | 14400 | mse | 752 KB | 4 s |
| be159137cb567588 | 64 | mse | 728 KB | 2 s |
| 62a4952532a62244 | 14400 | mae | 4524 KB | 22 s |
| 65aa5e34b2f35af1 | 3600 | none | 1700 KB | 34 s |

That is about 74 s of compilation in a 111 s run. The two main-model artifacts, the `mae` training kernel at 22 s and the 3600-row forward kernel at 34 s, account for 56 s of it. The three `mse` artifacts are the estimator surrogates and the device calibration probe.

Two things drive the size. The row count is baked into each kernel, so the same model compiles separately for the 14400-row training split and the 3600-row validation split. And `fp8` is emulated in software, so every arithmetic site expands into inline encode and decode blocks.

## Evidence that `fp8` is the driver, not the estimators

The largest artifact is 4,626,574 bytes and 104,976 lines of PTX. Its instruction mix is dominated by bit manipulation, `bra.uni` 6490, `or.b32` 6117, `setp.eq.b32` 5572, `and.b32` 5265, `shl.b32` 4726, with basic blocks named `subnormal`, `normal.pack` and `encode.normal`. Those are the software float codec. The XGBoost predictor contributes 19 `setp.gt.f` and 26 `selp.f64` instructions to that same file, so the tree ensemble is not what makes it large.

The same two-estimator shape at `fp(64)` costs a fraction of it. `issue-497c.rs` uses `.forest(2).elu().qi(2).k` plus `.xgbst().tan().qi(5).k` with `.fp(64)` and completes in 30.5 s cold and 5.1 s warm on the same die.

## Expected

A two-estimator model on a 18,000-row table should reach its first epoch in seconds. Over 100 seconds of native compilation for kernels whose epoch runs in 43 ms is a material defect in the `fp8` code path.

## Machine provenance

Discovered while verifying the #497 packets for PR #551 on archy nv0 (Tesla M60, die 3), `origin/minimal` at 10148e25. The failure reproduces identically on `origin/minimal` and on the #551 head, so it is not caused by that change.
